### PR TITLE
fix: adds fd-button--menu styling

### PR DESCRIPTION
**Pull request by InnaAtanasova**
## Related Issue
Closes SAP/fundamental-styles#336

## Description
added `--menu` modifier styling

## Screenshots
Screen Shot 2019-09-19 at 4 34 34 PM

### CODE REVIEW

File: scss/components/button.scss [hunk 1]
```diff
@@ -447,4 +448,102 @@ a.#{$block} {
...
+        }
+      }
+      &:active,
+      &.is-active,
+      &[aria-expanded="true"],
+      &:focus {
+        box-shadow: 0 0 0 1px var(--fd-color-status-3, map-get($fd-colors-status, 3));
+      }
+    }
+  }
```
Comment (greg-a-smith) [inline scss/components/button.scss]: This seems like a lot of unnecessary styling to just add a down arrow icon for `fd-button--dropdown`.  This is just a modifier and _should_ be able to co-exist with other base button styling and other elements and modifiers.

File: test/templates/button/index.njk [hunk 2]
```diff
@@ -94,4 +94,19 @@
...
+
+{% set example %}
+
+<button class="fd-button fd-button--dropdown">Dropdown</button>
+<br>
+<button class="fd-button fd-button--compact fd-button--dropdown">Dropdown</button>
+<br>
+<button class="fd-button fd-button--dropdown full-width">Dropdown</button>
+<br>
+<button class="fd-button fd-button--dropdown full-width error">Dropdown</button>
```
Comment (greg-a-smith) [inline test/templates/button/index.njk]: Don't "error" buttons just use the `fd-button--negative` class?

File: docs/pages/components/button.md [hunk 3]
```diff
@@ -204,3 +204,35 @@ Button with multiple actions
...
+  <br>
+   <button class="fd-button fd-button--compact fd-button--dropdown">Dropdown</button>
+  <br>
+  <button class="fd-button fd-button--dropdown" disabled>Dropdown</button>
+  <br>
+  <button class="fd-button fd-button--dropdown full-width">Dropdown</button>
+  <br>
+  <button class="fd-button fd-button--dropdown full-width error">Dropdown</button>
+  <br>
+  <button class="fd-button fd-button--dropdown full-width error" disabled>Dropdown</button>
```
Comment (greg-a-smith) [inline docs/pages/components/button.md]: Don't "error" buttons just use the `fd-button--negative` class?

File: test/templates/button/index.njk (hunk 2 above)
Comment (InnaAtanasova) [reply test/templates/button/index.njk]: Screen Shot 2019-09-19 at 1 53 14 PM
Screen Shot 2019-09-19 at 1 53 28 PM
Agreed, that was my original idea, but  `fd-button--negative` is a button with red text and red border, on hover it has red background and white color. In our case we only need a red border, so the amount of styling is less than overwriting the default behavior of `fd-button--negative`

**Pull request merged**